hangman/src/index.js

```diff
@@ -0,0 +1,288 @@
+mainContainerPart2.classList.add('main-container__part');
+mainContainerPart2.appendChild(game);
+
+// Append both parts to the main container
+mainContainer.appendChild(mainContainerPart1);
+mainContainer.appendChild(mainContainerPart2);
+
+// Append the main container to the body
+document.body.appendChild(mainContainer);
+
+const gallows = document.querySelector('.gallows-container');
+
+function displayPartOfBody(partsDrawn) {
+  const bodyPart = document.createElement('img');
+  bodyPart.classList.add('bodypart');
+  if (partsDrawn === 1) {
+    bodyPart.src = humanHead;
+    gallows.appendChild(bodyPart);
+  } else if (partsDrawn === 2) {
+    bodyPart.src = humanBody;
+    gallows.appendChild(bodyPart);
+  } else if (partsDrawn === 3) {
+    bodyPart.src = humanArmLeft;
+    gallows.appendChild(bodyPart);
+  } else if (partsDrawn === 4) {
+    bodyPart.src = humanArmRight;
+    gallows.appendChild(bodyPart);
+  } else if (partsDrawn === 5) {
+    bodyPart.src = humanLegLeft;
+    gallows.appendChild(bodyPart);
+  } else if (partsDrawn === 6) {
+    bodyPart.src = humanLegRight;
+    gallows.appendChild(bodyPart);
+  }
+}
+
+const url = './src/questions.json';
+let jsonData;
+try {
+  let response = await fetch(url);
+
+  if (response.ok) {
+    // если HTTP-статус в диапазоне 200-299, получаем тело ответа
+    jsonData = await response.json();
+  } else {
+    alert('Error: ' + response.status);
+  }
+} catch (error) {
+  // обработка ошибок при запросе
+  alert('Error: ' + error);
+}
+
+/* Game Constants */
+
+const maxWrongGuesses = 6;
+
+const wordDisplay = document.querySelector('#word');
+const hintDisplay = document.querySelector('#hint');
+const wrongGuessesDisplay = document.querySelector('#wrongGuesses');
+
+/* Game State */
+
+let word = pickRandom(jsonData);
+let rightGuesses = [];
+let wrongGuesses = [];
+let wrongGuessesCount = 0;
+
+/* Game Logic */
+
+function pickRandom(arr) {
+  return arr[Math.floor(Math.random() * arr.length)];
+}
+function resetGame() {
+  word = pickRandom(jsonData);
+  rightGuesses = [];
+  wrongGuesses = [];
+  wrongGuessesCount = 0;
+
+  clearKeyboard();
+  clearStatus();
+  hideBodyParts();
+  updateDisplay();
+}
+function updateDisplay() {
+  wordDisplay.innerHTML = word.word
+    .split('')
+    .map((letter) => (rightGuesses.includes(letter) ? letter : '__'))
+    .join(' ');
+
+  hintDisplay.innerHTML = `Hint: ${word.hint}`;
+  wrongGuessesDisplay.innerHTML = `Incorrect: ${wrongGuessesCount}/6`;
+}
+
+function clearKeyboard() {
+  const buttons = keyboard.querySelectorAll('.key');
+  for (const button of buttons) {
+    button.disabled = false;
+    button.classList.remove('correct');
+    button.classList.remove('incorrect');
+  }
+}
+function hideBodyParts() {
+  const bodyparts = document.querySelectorAll('.bodypart');
+  bodyparts.forEach((x) => x.remove());
+}
+function clearStatus() {
+  statusDisplay.innerHTML = '';
+}
+
+/* Init */
+
+createKeyboard();
```

Comment:
I would put this at the very bottom of the module after all function declarations. Now it's quite hard to find where your app really starts.